core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/meta/MetaExpression.kt

```diff
@@ -5,34 +5,47 @@ import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.Field
 import ch.ergon.dope.resolvable.fromable.AliasedBucket
 import ch.ergon.dope.resolvable.fromable.Bucket
+import ch.ergon.dope.resolvable.fromable.UnaliasedBucket
+import ch.ergon.dope.resolvable.operator.FunctionOperator
 import ch.ergon.dope.validtype.NumberType
 import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
 
-class MetaExpression(private val bucket: Bucket) : TypeExpression<StringType> {
+class MetaExpression(private val bucket: Bucket) : TypeExpression<StringType>, FunctionOperator {
     override fun toDopeQuery(): DopeQuery {
         val bucketDopeQuery = bucket.toDopeQuery()
         return DopeQuery(
-            queryString = "META(${
-            when (bucket) {
-                is AliasedBucket -> bucket.alias
-                else -> bucketDopeQuery.queryString
-            }
-            })",
+            queryString = toFunctionQueryString(
+                "META",
+                when (bucket) {
+                    is AliasedBucket -> "`${bucket.alias}`"
+                    is UnaliasedBucket -> bucketDopeQuery.queryString
+                },
+            ),
             parameters = bucketDopeQuery.parameters,
         )
     }
 
-    val cas: Field<NumberType> = Field("cas", toDopeQuery().queryString)
+    val cas: Field<NumberType> = getMetaField("cas")
 
-    val expiration: Field<NumberType> = Field("expiration", toDopeQuery().queryString)
+    val expiration: Field<NumberType> = getMetaField("expiration")
 
-    val flags: Field<NumberType> = Field("flags", toDopeQuery().queryString)
+    val flags: Field<NumberType> = getMetaField("flags")
 
-    val id: Field<StringType> = Field("id", toDopeQuery().queryString)
+    val id: Field<StringType> = getMetaField("id")
 
-    val type: Field<StringType> = Field("type", toDopeQuery().queryString)
+    val type: Field<StringType> = getMetaField("type")
 
-    val keyspace: Field<StringType> = Field("keyspace", toDopeQuery().queryString)
+    val keyspace: Field<StringType> = getMetaField("keyspace")
+
+    private fun <T : ValidType> getMetaField(field: String): MetaField<T> = MetaField(field, toDopeQuery().queryString)
```

Comment:
pass the whole dopeQuery and return the parameters as well

Reply:
done